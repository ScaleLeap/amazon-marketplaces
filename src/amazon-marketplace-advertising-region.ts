export interface AmazonMarketplaceAdvertisingRegion {
  /**
   * Region code.
   */
  code: string

  /**
   * Region name.
   */
  name: string

  /**
   * Authorization URI.
   */
  authorizationUri: string

  /**
   * Access Token URI
   */
  accessTokenUri: string

  /**
   * Access endpoint URI.
   */
  endpoint: string
}

export class AmazonMarketplaceAdvertisingRegion implements AmazonMarketplaceAdvertisingRegion {
  constructor(data: AmazonMarketplaceAdvertisingRegion) {
    Object.assign(this, data)
    Object.freeze(this)
  }
}
