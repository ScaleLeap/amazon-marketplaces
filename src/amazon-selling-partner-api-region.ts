export interface AmazonSellingPartnerApiRegion {
  /**
   * Region code.
   */
  code: string

  /**
   * Region name.
   */
  name: string

  /**
   * Access endpoint URI.
   */
  endpoint: string

  /**
   * AWS region.
   */
  awsRegion: string
}

export class AmazonSellingPartnerApiRegion implements AmazonSellingPartnerApiRegion {
  constructor(data: AmazonSellingPartnerApiRegion) {
    Object.assign(this, data)
    Object.freeze(this)
  }
}
