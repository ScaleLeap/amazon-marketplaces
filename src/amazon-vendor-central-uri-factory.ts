import { AmazonMarketplaceCountryCode } from './amazon-marketplace'

export function amazonVendorCentralUriFactory(countryCode: AmazonMarketplaceCountryCode): string {
  return `https://vendorcentral.amazon.${countryCode.toLowerCase()}`
}
