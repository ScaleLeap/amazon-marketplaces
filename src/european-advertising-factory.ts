import {
  AmazonMarketplaceAdvertisingCountryCode,
  AmazonMarketplaceAdvertisingTimeZone,
  AmazonMarketplaceAdvertising,
} from './amazon-marketplace'

export function europeanAdvertisingFactory(
  countryCode: AmazonMarketplaceAdvertisingCountryCode,
): AmazonMarketplaceAdvertising {
  return {
    uri: 'https://advertising-api-eu.amazon.com',
    countryCode,
    bids: {
      sponsoredBrands: {
        min: 10,
        max: 3900,
      },
      sponsoredProducts: {
        min: 2,
        max: 100000,
      },
    },
    timeZone: AmazonMarketplaceAdvertisingTimeZone.EUROPE_PARIS,
  }
}
