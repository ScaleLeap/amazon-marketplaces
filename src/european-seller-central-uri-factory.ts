export function europeanSellerCentralUriFactory(): string {
  return 'https://sellercentral-europe.amazon.com'
}
