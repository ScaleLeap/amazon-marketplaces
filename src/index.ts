export * from './amazon-marketplace'
export * from './marketplaces'
export * from './utils'
export * from './amazon-selling-partner-api-region'
export * from './selling-partner-api-regions'
