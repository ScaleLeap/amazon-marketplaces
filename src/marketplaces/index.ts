import { AE } from './ae'
import { AU } from './au'
import { BE } from './be'
import { BR } from './br'
import { CA } from './ca'
import { CN } from './cn'
import { DE } from './de'
import { EG } from './eg'
import { ES } from './es'
import { FR } from './fr'
import { GB } from './gb'
import { IN } from './in'
import { IT } from './it'
import { JP } from './jp'
import { MX } from './mx'
import { NL } from './nl'
import { PL } from './pl'
import { SA } from './sa'
import { SE } from './se'
import { SG } from './sg'
import { TR } from './tr'
import { US } from './us'

export const amazonMarketplaces = {
  /**
   * United Arab Emirates
   */
  AE,

  /**
   * Australia
   */
  AU,

  /**
   * Belgium
   */
  BE,

  /**
   * Brazil
   */
  BR,

  /**
   * Canada
   */
  CA,

  /**
   * China
   */
  CN,

  /**
   * Germany
   */
  DE,

  /**
   * Egypt
   */
  EG,

  /**
   * Spain
   */
  ES,

  /**
   * France
   */
  FR,

  /**
   * United Kingdom
   */
  GB,

  /**
   * India
   */
  IN,

  /**
   * Italy
   */
  IT,

  /**
   * Japan
   */
  JP,

  /**
   * Mexico
   */
  MX,

  /**
   * Netherlands
   */
  NL,

  /**
   * Poland
   */
  PL,

  /**
   * Singapore
   */
  SA,

  /**
   * Sweden
   */
  SE,

  /**
   * Singapore
   */
  SG,

  /**
   * Turkey
   */
  TR,

  /**
   * United States
   */
  US,
} as const

/**
 * An array of all of the `AmazonMarketplace` objects.
 */
export const amazonMarketplacesList = Object.values(amazonMarketplaces)
