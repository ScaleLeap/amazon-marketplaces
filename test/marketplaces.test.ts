import { readdirSync } from 'node:fs'
import path from 'node:path'

import { amazonMarketplaces } from '../src/marketplaces'

const countryCodes = readdirSync(path.join(__dirname, '../src/marketplaces'))
  .map((f) => f.split('.')[0])
  .filter((f) => f.match(/^[a-z]{2}$/))
  .map((f) => f.toUpperCase())

describe('marketplace', () => {
  it('has a known number of marketplaces', () => {
    expect.assertions(1)

    expect(countryCodes).toHaveLength(22)
  })

  describe.each(countryCodes)('%s', (countryCode) => {
    const code = countryCode as unknown as keyof typeof amazonMarketplaces

    it('should match the exports', () => {
      expect.assertions(1)

      expect(amazonMarketplaces[code].countryCode).toBe(code)
    })

    it('should match snapshot', () => {
      expect.assertions(1)

      expect(amazonMarketplaces[code]).toMatchSnapshot()
    })
  })

  it('object is read only', () => {
    expect.assertions(1)

    expect(() => Object.assign(amazonMarketplaces.CA, { id: 'a' })).toThrow(
      /Cannot assign to read only property/,
    )
  })
})
