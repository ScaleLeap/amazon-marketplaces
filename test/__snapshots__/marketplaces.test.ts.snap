// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marketplace AE should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 18400,
        "min": 40,
      },
      "sponsoredProducts": Object {
        "max": 18400,
        "min": 24,
      },
    },
    "countryCode": "AE",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Asia/Dubai",
  },
  "countryCode": "AE",
  "currency": "AED",
  "id": "A2VIGQ35RCS4UG",
  "name": "United Arab Emirates",
  "sellerCentralUri": "https://sellercentral.amazon.ae",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.ae/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.me/apps/authorize/consent",
  },
  "uri": "https://www.amazon.ae",
  "vendorCentralUri": "https://vendorcentral.amazon.me",
  "webServiceUri": "https://mws.amazonservices.ae",
}
`;

exports[`marketplace AU should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 7000,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 141000,
        "min": 10,
      },
    },
    "countryCode": "AU",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.jp/auth/o2/token",
      "authorizationUri": "https://apac.account.amazon.com/ap/oa",
      "code": "FE",
      "endpoint": "https://advertising-api-fe.amazon.com",
      "name": "Far East",
    },
    "timeZone": "Australia/Sydney",
  },
  "countryCode": "AU",
  "currency": "AUD",
  "id": "A39IBJ37TRP1C6",
  "name": "Australia",
  "sellerCentralUri": "https://sellercentral.amazon.com.au",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "us-west-2",
      "code": "FE",
      "endpoint": "https://sellingpartnerapi-fe.amazon.com",
      "name": "Far East",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.com.au/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.com.au/apps/authorize/consent",
  },
  "uri": "https://www.amazon.com.au",
  "vendorCentralUri": "https://vendorcentral.amazon.com.au",
  "webServiceUri": "https://mws.amazonservices.com.au",
}
`;

exports[`marketplace BR should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 20000,
        "min": 53,
      },
      "sponsoredProducts": Object {
        "max": 370000,
        "min": 7,
      },
    },
    "countryCode": "BR",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.com/auth/o2/token",
      "authorizationUri": "https://www.amazon.com/ap/oa",
      "code": "NA",
      "endpoint": "https://advertising-api.amazon.com",
      "name": "North America",
    },
    "timeZone": "America/Sao_Paulo",
  },
  "countryCode": "BR",
  "currency": "BRL",
  "id": "A2Q3Y263D00KWC",
  "name": "Brazil",
  "sellerCentralUri": "https://sellercentral.amazon.com.br",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "us-east-1",
      "code": "NA",
      "endpoint": "https://sellingpartnerapi-na.amazon.com",
      "name": "North America",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.com.br/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.com.br/apps/authorize/consent",
  },
  "uri": "https://www.amazon.com.br",
  "vendorCentralUri": "https://vendorcentral.amazon.com.br",
  "webServiceUri": "https://mws.amazonservices.com",
}
`;

exports[`marketplace CA should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 4900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "CA",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.com/auth/o2/token",
      "authorizationUri": "https://www.amazon.com/ap/oa",
      "code": "NA",
      "endpoint": "https://advertising-api.amazon.com",
      "name": "North America",
    },
    "timeZone": "America/Los_Angeles",
  },
  "countryCode": "CA",
  "currency": "CAD",
  "id": "A2EUQ1WTGCTBG2",
  "name": "Canada",
  "sellerCentralUri": "https://sellercentral.amazon.ca",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "us-east-1",
      "code": "NA",
      "endpoint": "https://sellingpartnerapi-na.amazon.com",
      "name": "North America",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.ca/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.ca/apps/authorize/consent",
  },
  "uri": "https://www.amazon.ca",
  "vendorCentralUri": "https://vendorcentral.amazon.ca",
  "webServiceUri": "https://mws.amazonservices.ca",
}
`;

exports[`marketplace CN should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "CN",
  "currency": "CNY",
  "id": "AAHKV2X7AFYLW",
  "name": "China",
  "uri": "https://www.amazon.cn",
  "webServiceUri": "https://mws.amazonservices.com.cn",
}
`;

exports[`marketplace DE should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "DE",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "DE",
  "currency": "EUR",
  "id": "A1PA6795UKMFR9",
  "name": "Germany",
  "sellerCentralUri": "https://sellercentral-europe.amazon.com",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral-europe.amazon.com/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.de/apps/authorize/consent",
  },
  "uri": "https://www.amazon.de",
  "vendorCentralUri": "https://vendorcentral.amazon.de",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace EG should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "EG",
  "currency": "EGP",
  "id": "ARBP9OOSHTCHU",
  "name": "Egypt",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "vendorCentralAuthUri": "https://vendorcentral.amazon.me/apps/authorize/consent",
  },
  "uri": "https://www.amazon.eg",
  "vendorCentralUri": "https://vendorcentral.amazon.me",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace ES should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "ES",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "ES",
  "currency": "EUR",
  "id": "A1RKKUPIHCS9HS",
  "name": "Spain",
  "sellerCentralUri": "https://sellercentral-europe.amazon.com",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral-europe.amazon.com/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.es/apps/authorize/consent",
  },
  "uri": "https://www.amazon.es",
  "vendorCentralUri": "https://vendorcentral.amazon.es",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace FR should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "FR",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "FR",
  "currency": "EUR",
  "id": "A13V1IB3VIYZZH",
  "name": "France",
  "sellerCentralUri": "https://sellercentral-europe.amazon.com",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral-europe.amazon.com/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.fr/apps/authorize/consent",
  },
  "uri": "https://www.amazon.fr",
  "vendorCentralUri": "https://vendorcentral.amazon.fr",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace GB should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3100,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "UK",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/London",
  },
  "countryCode": "GB",
  "currency": "GBP",
  "id": "A1F83G8C2ARO7P",
  "name": "United Kingdom",
  "sellerCentralUri": "https://sellercentral-europe.amazon.com",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral-europe.amazon.com/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.co.uk/apps/authorize/consent",
  },
  "uri": "https://www.amazon.co.uk",
  "vendorCentralUri": "https://vendorcentral.amazon.co.uk",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace IN should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 2000,
        "min": 1,
      },
      "sponsoredProducts": Object {
        "max": 5000,
        "min": 1,
      },
    },
    "countryCode": "IN",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Asia/Kolkata",
  },
  "countryCode": "IN",
  "currency": "INR",
  "id": "A21TJRUUN4KGV",
  "name": "India",
  "sellerCentralUri": "https://sellercentral.amazon.in",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.in/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.in/apps/authorize/consent",
  },
  "uri": "https://www.amazon.in",
  "vendorCentralUri": "https://vendorcentral.amazon.in",
  "webServiceUri": "https://mws.amazonservices.in",
}
`;

exports[`marketplace IT should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "DE",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Paris",
  },
  "countryCode": "IT",
  "currency": "EUR",
  "id": "APJ6JRA9NG5V4",
  "name": "Italy",
  "sellerCentralUri": "https://sellercentral-europe.amazon.com",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral-europe.amazon.com/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.it/apps/authorize/consent",
  },
  "uri": "https://www.amazon.it",
  "vendorCentralUri": "https://vendorcentral.amazon.it",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace JP should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 776000,
        "min": 1000,
      },
      "sponsoredProducts": Object {
        "max": 10000000,
        "min": 200,
      },
    },
    "countryCode": "JP",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.jp/auth/o2/token",
      "authorizationUri": "https://apac.account.amazon.com/ap/oa",
      "code": "FE",
      "endpoint": "https://advertising-api-fe.amazon.com",
      "name": "Far East",
    },
    "timeZone": "Asia/Tokyo",
  },
  "countryCode": "JP",
  "currency": "JPY",
  "id": "A1VC38T7YXB528",
  "name": "Japan",
  "sellerCentralUri": "https://sellercentral.amazon.co.jp",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "us-west-2",
      "code": "FE",
      "endpoint": "https://sellingpartnerapi-fe.amazon.com",
      "name": "Far East",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.co.jp/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.co.jp/apps/authorize/consent",
  },
  "uri": "https://www.amazon.co.jp",
  "vendorCentralUri": "https://vendorcentral.amazon.co.jp",
  "webServiceUri": "https://mws.amazonservices.jp",
}
`;

exports[`marketplace MX should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 50000,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 2000000,
        "min": 10,
      },
    },
    "countryCode": "MX",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.com/auth/o2/token",
      "authorizationUri": "https://www.amazon.com/ap/oa",
      "code": "NA",
      "endpoint": "https://advertising-api.amazon.com",
      "name": "North America",
    },
    "timeZone": "America/Los_Angeles",
  },
  "countryCode": "MX",
  "currency": "MXN",
  "id": "A1AM78C64UM0Y8",
  "name": "Mexico",
  "sellerCentralUri": "https://sellercentral.amazon.com.mx",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "us-east-1",
      "code": "NA",
      "endpoint": "https://sellingpartnerapi-na.amazon.com",
      "name": "North America",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.com.mx/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.com.mx/apps/authorize/consent",
  },
  "uri": "https://www.amazon.com.mx",
  "vendorCentralUri": "https://vendorcentral.amazon.com.mx",
  "webServiceUri": "https://mws.amazonservices.com.mx",
}
`;

exports[`marketplace NL should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "NL",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Amsterdam",
  },
  "countryCode": "NL",
  "currency": "EUR",
  "id": "A1805IZSGTT6HS",
  "name": "Netherlands",
  "sellerCentralUri": "https://sellercentral.amazon.nl",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.nl/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.nl/apps/authorize/consent",
  },
  "uri": "https://www.amazon.nl",
  "vendorCentralUri": "https://vendorcentral.amazon.nl",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace PL should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "PL",
  "currency": "PLN",
  "id": "A1C3SOZRARQ6R3",
  "name": "Poland",
  "sellerCentralUri": "https://sellercentral.amazon.pl",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.pl/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.pl/apps/authorize/consent",
  },
  "uri": "https://www.amazon.pl",
  "vendorCentralUri": "https://vendorcentral.amazon.pl",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace SA should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "SA",
  "currency": "SAR",
  "id": "A17E79C6D8DWNP",
  "name": "Saudi Arabia",
  "sellerCentralUri": "https://sellercentral.amazon.sa",
  "uri": "https://www.amazon.sa",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace SE should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 940000000,
        "min": 900,
      },
      "sponsoredProducts": Object {
        "max": 930000,
        "min": 18,
      },
    },
    "countryCode": "SE",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.uk/auth/o2/token",
      "authorizationUri": "https://eu.account.amazon.com/ap/oa",
      "code": "EU",
      "endpoint": "https://advertising-api-eu.amazon.com",
      "name": "Europe",
    },
    "timeZone": "Europe/Stockholm",
  },
  "countryCode": "SE",
  "currency": "SEK",
  "id": "A2NODRKZP88ZB9",
  "name": "Sweden",
  "sellerCentralUri": "https://sellercentral.amazon.se",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.se/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.se/apps/authorize/consent",
  },
  "uri": "https://www.amazon.se",
  "vendorCentralUri": "https://vendorcentral.amazon.se",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace SG should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 10000,
        "min": 14,
      },
      "sponsoredProducts": Object {
        "max": 110000,
        "min": 2,
      },
    },
    "countryCode": "SG",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.co.jp/auth/o2/token",
      "authorizationUri": "https://apac.account.amazon.com/ap/oa",
      "code": "FE",
      "endpoint": "https://advertising-api-fe.amazon.com",
      "name": "Far East",
    },
    "timeZone": "Asia/Singapore",
  },
  "countryCode": "SG",
  "currency": "SGD",
  "id": "A19VAU5U5O7RUS",
  "name": "Singapore",
  "sellerCentralUri": "https://sellercentral.amazon.sg",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "us-west-2",
      "code": "FE",
      "endpoint": "https://sellingpartnerapi-fe.amazon.com",
      "name": "Far East",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.sg/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.com.sg/apps/authorize/consent",
  },
  "uri": "https://www.amazon.sg",
  "vendorCentralUri": "https://vendorcentral.amazon.com.sg",
  "webServiceUri": "https://mws-fe.amazonservices.com",
}
`;

exports[`marketplace TR should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "TR",
  "currency": "TRY",
  "id": "A33AVAJ2PDY3EV",
  "name": "Turkey",
  "sellerCentralUri": "https://sellercentral.amazon.com.tr",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "eu-west-1",
      "code": "EU",
      "endpoint": "https://sellingpartnerapi-eu.amazon.com",
      "name": "Europe",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.com.tr/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.com.tr/apps/authorize/consent",
  },
  "uri": "https://www.amazon.com.tr",
  "vendorCentralUri": "https://vendorcentral.amazon.com.tr",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace US should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 4900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "US",
    "region": AmazonMarketplaceAdvertisingRegion {
      "accessTokenUri": "https://api.amazon.com/auth/o2/token",
      "authorizationUri": "https://www.amazon.com/ap/oa",
      "code": "NA",
      "endpoint": "https://advertising-api.amazon.com",
      "name": "North America",
    },
    "timeZone": "America/Los_Angeles",
  },
  "countryCode": "US",
  "currency": "USD",
  "id": "ATVPDKIKX0DER",
  "name": "United States",
  "sellerCentralUri": "https://sellercentral.amazon.com",
  "sellingPartner": Object {
    "region": AmazonSellingPartnerApiRegion {
      "awsRegion": "us-east-1",
      "code": "NA",
      "endpoint": "https://sellingpartnerapi-na.amazon.com",
      "name": "North America",
    },
    "sellerCentralAuthUri": "https://sellercentral.amazon.com/apps/authorize/consent",
    "vendorCentralAuthUri": "https://vendorcentral.amazon.com/apps/authorize/consent",
  },
  "uri": "https://www.amazon.com",
  "vendorCentralUri": "https://vendorcentral.amazon.com",
  "webServiceUri": "https://mws.amazonservices.com",
}
`;
